src/main/java/seedu/address/commons/util/IcsUtil.java

```diff
@@ -0,0 +1,175 @@
+        String starttime = "";
+        String enddate = "";
+        String endtime = "";
+        String name = "";
+        boolean isTimeSlot = false;
+
+        for (String line : linesInChunk) {
+            String[] splitLine = line.split(":");
+            if (splitLine.length != 2) {
+                continue; //file format is wrong, just ignore the line
+            }
+
+            String dataName = splitLine[0];
+            String dataString = splitLine[1];
+
+            if (dataName.equals("DTSTART")) {
+                //get start
+                try {
+                    startdate = dataString.substring(0, 8);
+                    starttime = dataString.substring(9, 15);
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DataConversionException(e);
+                }
+            } else if (dataName.equals("DTEND")) {
+                //get end
+                try {
+                    enddate = dataString.substring(0, 8);
+                    endtime = dataString.substring(9, 15);
+                } catch (IndexOutOfBoundsException e) {
+                    throw new DataConversionException(e);
+                }
+            } else if (dataName.equals("SUMMARY")) {
+                //get mod name
+                name = dataString;
+            } else if (dataName.equals("RRULE")) {
+                //is it a weekly time slot?
+                isTimeSlot = dataString.contains("WEEKLY");
+            }
+        }
+
+        //check data is all present
+        if ((startdate.isEmpty()) || (enddate.isEmpty()) || (starttime.isEmpty())
+                || (endtime.isEmpty()) || (name.isEmpty()) || (!isTimeSlot)) {
+            return Optional.empty();
+        }
+
+        //process the data we got from the chunk
+        LocalTime timeSlotStartTime;
+        LocalTime timeSlotEndTime;
+        DayOfWeek timeSlotDay;
+
+        try {
+            timeSlotStartTime = icsStringToTime(starttime);
+            timeSlotEndTime = icsStringToTime(endtime);
+            timeSlotDay = icsStringToDay(startdate);
+        } catch (NumberFormatException e) {
+            throw new DataConversionException(e);
+        }
+
+        TimeSlot timeSlot = new TimeSlot(timeSlotDay, timeSlotStartTime, timeSlotEndTime);
+        return Optional.of(timeSlot);
+    }
+
+    /**
+     * Converts the date in the ics file into day (1-7).
+     */
+    private static DayOfWeek icsStringToDay(String dateString) {
+        DateTimeFormatter fmt = DateTimeFormatter.ofPattern("yyyyMMdd");
+        LocalDate date = LocalDate.parse(dateString, fmt);
+        DayOfWeek day = date.getDayOfWeek();
+
+        return day;
+    }
+
+    /**
+     * Converts the ics-formatted string into a LocalTime object.
+     */
+    private static LocalTime icsStringToTime(String icsTime) {
+        int timeInt = Integer.parseInt(icsTime);
+        timeInt += 80000; //need to add 8 hours to the ics-formatted version.
```

Comment:
my initial thoughts was that because NUSMODS's timetable starts at 0800, and the 8 hours offset is simply a quirky result of that. But now that you mention it makes sense! Will work on this for next PR.